OtpRetrieval.js

```diff
@@ -0,0 +1,80 @@
+import { NativeEventEmitter, NativeModules } from 'react-native'
+import React, { Component } from 'react'
+import GenericTextInput from './GenericTextInput';
+import R from 'ramda'
+import { Sentry } from 'react-native-sentry'
+import PropTypes from 'prop-types'
+
+const isEmpty = R.anyPass([R.isNil, R.isEmpty])
+
+const { SMSRetrieverModule } = NativeModules
+ export default class OtpRetrieval extends Component{
+
+  static propTypes = {
+    refCallback:PropTypes.func,
+    onAutoReadComplete:PropTypes.func,
+    position:PropTypes.number
+  }
+
+  static defaultProps = {
+    refCallback: () => {},
+    onAutoReadComplete: () => {},
+    position: 2
+  }
+
+    componentDidMount(){
+        this.startSMSListener()
+    }
+
+    extractOTPFromMessage = (message) => {
+      const { position } = this.props
+      let array = []
+      if (!isEmpty(message)) {
+        array = message.split(' ')
+      }
+    
+      if (!isEmpty(array) && array.length >= 3) {
+        return array[position]
+      }
+      return null
+    }
+
+    onOtpRecieved = (event) => {
+        if (!isEmpty(event)) {
+          const otp = this.extractOTPFromMessage(event.message)
+          this.otpRef.onChangeText(otp)
+          this.otpRetrievalComplete= true
+        }
+      }
+
+      onChangeOtpText = (otp) => {
+        this.props.onAutoReadComplete(otp)   
+      }
+
+      setOTPRef = (ref) => {
+        this.props.refCallback(ref)
+        this.otpRef = ref
+      }
+    
+      startSMSListener() {
+        try {
+          SMSRetrieverModule.startSMSListener()
+          const messageEventEmitter = new NativeEventEmitter(SMSRetrieverModule)
+          messageEventEmitter.addListener('com.RNSmsRetriever:otpReceived', this.onOtpRecieved)
+        } catch (exception) {
+          Sentry.captureException(exception)
+        }
+      }
+
+    render() {
+        const { position, onChangeOtpText, onChangeText , ...others} = this.props
```

Comment:
remove refCallback and onAutoReadComplete from the props. Why are we removing onChangeOtpText?